app.py

```diff
@@ -75,112 +94,264 @@ def toDict(responses, baseQuestions=None):
             responseAsDict[partie_avant] = partie_apres
     return responseAsDict
 
+# Function to read CSV file
 def read_csv_file(file_path):
-    # Lire le fichier CSV
+    # Read csv file
     df = pd.read_csv(file_path, sep=';')
     return df
 
-def  create_final_request(file, baseQuestions, typeOfQuestion):
+# Function to create base request
+def create_base_request(file):
+    requestAsText = ""
+    for index, line in file.iterrows():
+        if not line.isnull().all():
+            specification = line['Specification']
+            requestAsText += str(specification) + "\n"
+
+        line_text = str(line['Key']) + ":" + str(line['Question']) + "; \n"
+        requestAsText += line_text
+    return requestAsText
+
+# Function to create final request
+def  create_final_request(file, baseQuestions):
     request_AsText = ""
     for index, line in file.iterrows():
         if not line.isnull().all():
             categorie = str(line['Categories'])
+
             if line['Specification'] != None:
                 request_AsText += str(line['Specification']) + "\n"
 
             if line['Categories'] != None:
-                if "not_asked" in categorie:
-                    break
-                elif "all" in categorie:
-                    request_AsText = addLine(request_AsText, line, typeOfQuestion)
+                if "all" in categorie:
+                    request_AsText = add_line(request_AsText, line)
+
+                if  "fertilizer"in categorie and"is_fertilizer" in baseQuestions:
+                    if "contain_pesticide" in baseQuestions:
+                        if "pesticide" in categorie:
+                            request_AsText = add_line(request_AsText, line)
+
+                    if "is_seed" in baseQuestions:
```

Comment:
I think you can use `and` here to simply the if

```suggestion
                    if "is_seed" in baseQuestions and "seed" in categorie:
```

Reply:
No i can because is_seed can appear in multiple situation i code this as if in the futur we put multiple question